pkg/recommendation/minmaxcapacity.go

```diff
@@ -0,0 +1,203 @@
+		// like this -
+		// Node-1
+		//     - [100GB, 4]
+		//     - [500GB, 6]
+		//     - [1TB, 4]
+		// Node-2
+		//     - [100GB, 1]
+		//     - [500GB, 8]
+		//     - [1TB, 2]
+		// Node-3
+		//     - [500GB, 5]
+		//     - [1TB, 5]
+		// Node-4
+		//     - [100GB, 8]
+		//     - [500GB, 3]
+		nodeCapacityDeviceCountMap := make(map[string]map[int64]int64)
+
+		// Iterating over nodeBlockDeviceListMap map and populating
+		// nodeCapacityDeviceCountMap
+		for nodeName, blockDeviceList := range nodeBlockDeviceListMap {
+
+			// If capacity count map not found for a node then create it.
+			capacityCountMap, capacityCountMapFound := nodeCapacityDeviceCountMap[nodeName]
+			if !capacityCountMapFound {
+				capacityCountMap = make(map[int64]int64)
+			}
+
+			// Iterate over blockDeviceList and populate
+			for _, blockDevice := range blockDeviceList {
+
+				// find capacity for each block device nodeBlockDeviceListMap
+				capacity, capacityFound := blockDevice.Capacity.AsInt64()
+				if !capacityFound {
+					// TODO handle this case
+					continue
+				}
+
+				// In capacity device count map if entry not found for any capacity then
+				// make it 0. If found then increase by 1.
+				_, ok := capacityCountMap[capacity]
+				if !ok {
+					capacityCountMap[capacity] = 0
+				}
+				capacityCountMap[capacity]++
+			}
+
+			// update nodeCapacityDeviceCountMap for each node.
+			nodeCapacityDeviceCountMap[nodeName] = capacityCountMap
+		}
+
+		result := &MinMaxCapacity{
+			MinCapacity: resource.Quantity{},
+			MaxCapacity: resource.Quantity{},
+		}
+
+		// Iterate over nodeCapacityDeviceCountMap and get min and max.
+		for _, capacityDeviceCountMap := range nodeCapacityDeviceCountMap {
+			for capacity, deviceCount := range capacityDeviceCountMap {
+
+				// If device count is less than group device count
+				// then skip those block devices.
+				if deviceCount < r.RaidGroupConfig.GroupDeviceCount {
+					continue
+				}
+
+				// effectiveBlockDeviceCount is round up block device count.
+				// ie - one node has 5 block devices of 100GB and pool type
+				// is mirror then effective block device count will be 4.
+				effectiveBlockDeviceCount := deviceCount / r.RaidGroupConfig.GroupDeviceCount
+
+				// Get min capacity from block device capacity and count.
+				// If min capacity in response 0 or greater than new min
+				// then update it.
+				min, err := resource.ParseQuantity(fmt.Sprintf("%d",
+					(capacity * r.RaidGroupConfig.GetDataDeviceCount())))
+				if err != nil {
+					continue
+				}
+				if result.MinCapacity.IsZero() || result.MinCapacity.Cmp(min) > 0 {
+					result.MinCapacity = min
+				}
+
+				// Get max capacity from block device capacity and count.
+				// If max capacity in response 0 or less than new max then
+				// update it.
+				max, err := resource.ParseQuantity(fmt.Sprintf("%d",
+					(capacity * (effectiveBlockDeviceCount / r.RaidGroupConfig.GroupDeviceCount) *
```

Comment:
This needs to be updated as per the above comment on effectiveBlockDeviceCount.

Reply:
In that case, value will be
capacity * (effectiveBlockDeviceCount / r.RaidGroupConfig.GroupDeviceCount)